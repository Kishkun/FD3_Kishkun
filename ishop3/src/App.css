*:focus {
  box-sizing: border-box;
  margin: 0;
  outline: none
}
a {
  outline: none;
  text-decoration: none;
  color: #2B2A29;
}
li {
  list-style-type: none;
}
ul {
  padding: 0;
}
html,
body {
  padding: 0;
  min-height: 100vh;
  font-family: 'Lato', sans-serif;
  font-size: 14px;
  max-width: 1920px;
  margin: 0 auto;
  color: #2B2A29;
  background-color: #ffffff;
  /*400 700 900*/
}
.shop_wrapper {
  max-width: 1200px;
  margin: 0 auto;
  padding-top: 60px;
}
.shop_wrapper .shop_title {
  text-align: center;
  font-size: 3vw;
  font-weight: 700;
  margin-bottom: 40px;
}
.mobile_bg {
  background: linear-gradient(122.04deg,#14e9b6 23.38%,#076deb 82.75%);
}