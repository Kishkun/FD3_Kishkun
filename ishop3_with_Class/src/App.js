import React from 'react';
import PropTypes from 'prop-types';
import Mobile from './components/mobile/Mobile';
import Header from './components/header/Header';
import Footer from './components/footer/Footer';

class App extends React.Component {

    state = {
        mobileList: this.props.mobiles,
    };

    deleteItem = (id) => {
        let filteredList = this.state.mobileList.filter(item => {
            return item.id !== id
        });
        this.setState({mobileList: filteredList});
        console.log(id);
    };

    render() {
        return (
            <div className="shop_wrapper">
                <Header title={this.props.title}/>
                <main>
                    <Mobile mobiles={this.state.mobileList}
                            deleteItem={this.deleteItem}
                    />
                </main>
                <Footer/>
            </div>
        );
    }
}

App.defaultProps = {
    title: 'Online store',
    mobiles: []
};

App.propTypes = {
    title: PropTypes.string.isRequired,
    mobiles: PropTypes.array.isRequired,
};

export default App;
