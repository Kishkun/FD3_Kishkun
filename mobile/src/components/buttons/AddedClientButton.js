import React from 'react';
import myEvents from './../../events';

const AddedClientButton = (props) => {
    console.log('add button function');
    return (
        <button
            type='button'
            className='clients_btn add_btn'
            onClick={()=> myEvents.emit('onOpenAddedItem', this)}
        >Добавить клиента</button>
    )
};

export default AddedClientButton;