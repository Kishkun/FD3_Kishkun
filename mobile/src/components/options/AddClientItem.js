import React from 'react';
import PropTypes from 'prop-types';
import myEvents from '../../events';

class AddClientItem extends React.PureComponent {

    static propTypes = {};

    constructor(props) {
        super(props);
        this.state = {
            inputs: {
                firstName: '',
                secondName: '',
                patronymic: '',
                balance: '',
                status: true
            },
            errors: {
                firstName: false,
                secondName: false,
                patronymic: false,
                balance: false,
            },
            isDisabled: false
        };
        this.firstNameRef = React.createRef();
        this.secondNameRef = React.createRef();
        this.patronymicRef = React.createRef();
        this.balanceRef = React.createRef();
    }

    addSubmit = (e) => {
        e.preventDefault();
        let newItem = {
            firstName: this.firstNameRef.current.value,
            secondName: this.secondNameRef.current.value,
            patronymic: this.patronymicRef.current.value,
            balance: +this.balanceRef.current.value,
            status: this.state.inputs.status,
        };
        if (newItem.firstName && newItem.secondName && newItem.patronymic && newItem.balance) {
            myEvents.emit('onAddItem', newItem)
        }
        this.setState({
            inputs: {
                firstName: '',
                secondName: '',
                patronymic: '',
                balance: '',
            },
            isDisabled: true
        });
    };


    validationCheck = (item) => {
        if (item.current.value.length === 0) {
            this.setState({
                isDisabled: true,
                errors: {
                    ...this.state.errors,
                    [item.current.id] : true
                }
            })
        } else {
            this.setState({
                isDisabled: false,
                errors: {
                    ...this.state.errors,
                    [item.current.id] : false
                }
            })
        }
        if ( item === this.balanceRef) {
            this.validateBalance(item);
        }
    };

    validateBalance(item) {
        if (item.current.value > 0) {
            this.setState({
                inputs: {
                    status: true
                }
            })
        } else {
            this.setState({
                inputs: {
                    status: false
                }
            })
        }
    }

    render() {
        let {inputs, errors, isDisabled} = this.state;
        return (
            <form onSubmit={this.addSubmit} className='add_client_form'>
                <ul className='list'>
                    <li className='list_item'>
                        <label htmlFor='firstName'>
                            <input type='text'
                                   id='firstName'
                                   name='firstName'
                                   className={!errors.firstName ? 'form_input' : 'form_input error'}
                                   ref={this.firstNameRef}
                                   defaultValue={inputs.firstName}
                                   onBlur={() => this.validationCheck(this.firstNameRef)}
                            />
                        </label>
                    </li>
                    <li className='list_item'>
                        <label htmlFor='secondName'>
                            <input type='text'
                                   id='secondName'
                                   name='secondName'
                                   className={!errors.secondName ? 'form_input' : 'form_input error'}
                                   ref={this.secondNameRef}
                                   defaultValue={inputs.secondName}
                                   onBlur={() => this.validationCheck(this.secondNameRef)}
                            />
                        </label>
                    </li>
                    <li className='list_item'>
                        <label htmlFor='patronymic'>
                            <input type='text'
                                   id='patronymic'
                                   name='patronymic'
                                   className={!errors.patronymic ? 'form_input' : 'form_input error'}
                                   ref={this.patronymicRef}
                                   defaultValue={inputs.patronymic}
                                   onBlur={() => this.validationCheck(this.patronymicRef)}
                            />
                        </label>
                    </li>
                    <li className='list_item'>
                        <label htmlFor='balance'>
                            <input type='number'
                                   id='balance'
                                   name='balance'
                                   className={!errors.balance ? 'form_input' : 'form_input error'}
                                   ref={this.balanceRef}
                                   defaultValue={inputs.balance}
                                   onBlur={() => this.validationCheck(this.balanceRef)}
                            />
                        </label>
                    </li>
                </ul>
                <div className='form_btn_block'>
                    <button
                        type='submit'
                        className={!isDisabled ? 'clients_btn save_btn' : 'clients_btn save_btn disabled'}>
                        save
                    </button>
                    <button
                        type='button'
                        className='clients_btn cancel_btn'
                        onClick={() => myEvents.emit('closeForm', this)}
                    >
                        cancel
                    </button>
                </div>
            </form>
        );
    }
}

export default AddClientItem;
