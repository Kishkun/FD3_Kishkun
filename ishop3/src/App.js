import React from 'react';
import PropTypes from 'prop-types';
import './App.css';
import Mobile from './components/mobile/Mobile';

let App = (props) => {
    return (
        <div className="shop_wrapper">
            <header>
                <h1 className='shop_title'>{props.title}</h1>
            </header>
            <Mobile mobiles={props.mobiles}
                    removeItem={props.removeItem}
                    toggleBackgroundItem={props.toggleBackgroundItem}
            />
        </div>
    );
};

App.defaultProps = {
    title: 'Online store',
    mobiles: []
};

App.propTypes = {
    title: PropTypes.string.isRequired,
    mobiles: PropTypes.array.isRequired,
};

export default App;
