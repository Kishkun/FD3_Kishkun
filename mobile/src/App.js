import React from 'react';
import PropTypes from 'prop-types';
import myEvents from './events';
import CompanyName from './components/CompanyName';
import MobileClients from './components/clients/MobileClients';
import AddedClientButton from './components/buttons/AddedClientButton';
import FilterStatusButtons from './components/buttons/FilterStatusButtons';
import EditClientItem from './components/options/EditClientItem';
import AddClientItem from './components/options/AddClientItem';

class App extends React.PureComponent {

    static propTypes = {
        title: PropTypes.string.isRequired,
        companies: PropTypes.arrayOf(
            PropTypes.shape({
                id: PropTypes.number.isRequired,
                name: PropTypes.string.isRequired
            })
        ),
        clients:PropTypes.arrayOf(
            PropTypes.shape({
                id: PropTypes.number.isRequired,
                firstName: PropTypes.string.isRequired,
                secondName: PropTypes.string.isRequired,
                patronymic: PropTypes.string.isRequired,
                balance: PropTypes.number.isRequired,
                status: PropTypes.bool.isRequired
            })
        ),
    };

    constructor(props) {
        super(props);
        this.state = {
            filter: 'all', //active, all, blocked,
            clients: this.props.clients,
            companies: this.props.companies,
            title: this.props.title,
            editingItem: null,
            isShowAddedForm: false
        };
        this.minID = this.props.clients.length;
        myEvents.on('onFilterChange', this.onFilterChange.bind(this));
        myEvents.on('onDeleteItem', this.onDeleteItem.bind(this));
        myEvents.on('onEditItem', this.onEditItem.bind(this));
        myEvents.on('closeForm', this.closeForm.bind(this));
        myEvents.on('onClientChange', this.onClientChange.bind(this));
        myEvents.on('onOpenAddedItem', this.onOpenAddedItem.bind(this));
        myEvents.on('onAddItem', this.onAddItem.bind(this));
    }

    minID = 1;

    createId = (item) => {
        return {
            id: ++this.minID,
            ...item
        };
    };

    filter(clients, filter) {
        switch (filter) {
            case 'all':
                return clients;
            case 'active':
                return clients.filter((item) => item.status);
            case 'blocked':
                return clients.filter((item) => !item.status);
            default:
                return clients;
        }
    }

    onDeleteItem = (id) => {
        let filteredList = this.state.clients.filter(item => item.id !== id);
        this.setState({clients: filteredList});
    };

    onFilterChange = (filter) => {
        this.setState({filter: filter})
    };

    onEditItem = (id) => {
        let editingItem = this.state.clients.find(item => item.id === id);
        this.setState({editingItem: editingItem});
    };

    onClientChange = (client) => {
        this.setState((state) =>{
            let findClient = state.clients.findIndex((el) => el.id === client.id);
            let newClients = [...state.clients];
            newClients[findClient] = client;
            return {
                clients: newClients,
                editingItem: null
            }
        });
    };

    onOpenAddedItem = () => {
        this.setState({isShowAddedForm: true});
    };

    closeForm = () =>{
        this.setState({editingItem: null, isShowAddedForm: false});
    };

    onAddItem = (item) => {
        let newItem = this.createId(item);
        let newList = [...this.state.clients, newItem];
        this.setState({
            clients: newList,
            isShowAddedForm: false,
        });
    };

    render() {
        console.log("MobileCompany render");
        let {filter, clients, companies, title} = this.state;
        let filtered = this.filter(clients, filter);
        return (
            <div className='company_wrapper'>
                <h1 className='company_title'>{title}</h1>
                <CompanyName companies={companies} />
                <hr/>
                <FilterStatusButtons filter={filter} />
                <hr/>
                <MobileClients
                    clients={filtered}/>
                <hr/>
                <AddedClientButton/>
                {this.state.isShowAddedForm ? <AddClientItem/> : false}
                {this.state.editingItem ? <EditClientItem
                    key={this.state.editingItem.id}
                    editingItem={this.state.editingItem} /> : null}
            </div>
        );
    }
}

export default App;
