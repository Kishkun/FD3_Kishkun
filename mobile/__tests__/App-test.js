import React from 'react';
import renderer from 'react-test-renderer';
import myEvents from '../src/events';
import App from '../../mobile/src/App';
import AddedClientButton from '../src/components/buttons/AddedClientButton';
import AddClientItem from '../src/components/options/AddClientItem';

test('тестирование добавление клиента', () => {

    let clients = [];
    let title = 'Test App';
    let companies = [
        {
            'id': 1,
            'name': 'Velcom'
        }
    ];

    const component = renderer.create(
        <App clients={clients}
             title={title}
             companies={companies}/>
    );

    let componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();


    let addButton = component.root.findByType(AddedClientButton).findByType('button');

    addButton.props.onClick();

    componentTree = component.toJSON();
    expect(componentTree).toMatchSnapshot();


    let newItem = {
        firstName: 'Alex',
        secondName: 'Kishkun',
        patronymic: 'Vladimirovich',
        balance: 1000,
        status: true
    };
    // let input = component.root.findByType(AddClientItem).findAllByType('input');
    //
    // input[0].value = 'alex';
    // console.log( input[1].value);
    // let addForm = component.root.findByType(AddClientItem).findByType('form');
    // addForm.props.onSubmit();

    // let closeButton = component.root.findByType(AddClientItem).findAllByType('button');
    // console.log(closeButton[1]);
    //
    // closeButton[1].props.onClick();
    //
    // componentTree = component.toJSON();
    // expect(componentTree).toMatchSnapshot();
});