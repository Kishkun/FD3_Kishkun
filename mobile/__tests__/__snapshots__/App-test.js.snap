// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`тестирование добавление клиента 1`] = `
<div
  className="company_wrapper"
>
  <h1
    className="company_title"
  >
    Test App
  </h1>
  <div
    className="name_block"
  >
    <button
      className="name_btn active"
      onClick={[Function]}
      type="button"
    >
      Velcom
    </button>
    <p
      className="name"
    >
      Компания: 
      Velcom
    </p>
  </div>
  <hr />
  <div
    className="filter_block"
  >
    <button
      className="clients_btn active"
      onClick={[Function]}
      type="button"
    >
      Все
    </button>
    <button
      className="clients_btn"
      onClick={[Function]}
      type="button"
    >
      Активные
    </button>
    <button
      className="clients_btn"
      onClick={[Function]}
      type="button"
    >
      Заблокированые
    </button>
  </div>
  <hr />
  <div
    className="clients_wrapper"
  >
    <div
      className="clients_title"
    >
      <div
        className="th"
      >
        Фамилия
      </div>
      <div
        className="th"
      >
        Имя
      </div>
      <div
        className="th"
      >
        Отчество
      </div>
      <div
        className="th"
      >
        Баланс
      </div>
      <div
        className="th"
      >
        Статус
      </div>
      <div
        className="th button_th"
      >
        Редактировать
      </div>
      <div
        className="th button_th"
      >
        Удалить
      </div>
    </div>
  </div>
  <hr />
  <button
    className="clients_btn add_btn"
    onClick={[Function]}
    type="button"
  >
    Добавить клиента
  </button>
</div>
`;

exports[`тестирование добавление клиента 2`] = `
<div
  className="company_wrapper"
>
  <h1
    className="company_title"
  >
    Test App
  </h1>
  <div
    className="name_block"
  >
    <button
      className="name_btn active"
      onClick={[Function]}
      type="button"
    >
      Velcom
    </button>
    <p
      className="name"
    >
      Компания: 
      Velcom
    </p>
  </div>
  <hr />
  <div
    className="filter_block"
  >
    <button
      className="clients_btn active"
      onClick={[Function]}
      type="button"
    >
      Все
    </button>
    <button
      className="clients_btn"
      onClick={[Function]}
      type="button"
    >
      Активные
    </button>
    <button
      className="clients_btn"
      onClick={[Function]}
      type="button"
    >
      Заблокированые
    </button>
  </div>
  <hr />
  <div
    className="clients_wrapper"
  >
    <div
      className="clients_title"
    >
      <div
        className="th"
      >
        Фамилия
      </div>
      <div
        className="th"
      >
        Имя
      </div>
      <div
        className="th"
      >
        Отчество
      </div>
      <div
        className="th"
      >
        Баланс
      </div>
      <div
        className="th"
      >
        Статус
      </div>
      <div
        className="th button_th"
      >
        Редактировать
      </div>
      <div
        className="th button_th"
      >
        Удалить
      </div>
    </div>
  </div>
  <hr />
  <button
    className="clients_btn add_btn"
    onClick={[Function]}
    type="button"
  >
    Добавить клиента
  </button>
  <form
    className="add_client_form"
    onSubmit={[Function]}
  >
    <ul
      className="list"
    >
      <li
        className="list_item"
      >
        <label
          htmlFor="firstName"
        >
          <input
            className="form_input"
            defaultValue=""
            id="firstName"
            name="firstName"
            onBlur={[Function]}
            type="text"
          />
        </label>
      </li>
      <li
        className="list_item"
      >
        <label
          htmlFor="secondName"
        >
          <input
            className="form_input"
            defaultValue=""
            id="secondName"
            name="secondName"
            onBlur={[Function]}
            type="text"
          />
        </label>
      </li>
      <li
        className="list_item"
      >
        <label
          htmlFor="patronymic"
        >
          <input
            className="form_input"
            defaultValue=""
            id="patronymic"
            name="patronymic"
            onBlur={[Function]}
            type="text"
          />
        </label>
      </li>
      <li
        className="list_item"
      >
        <label
          htmlFor="balance"
        >
          <input
            className="form_input"
            defaultValue=""
            id="balance"
            name="balance"
            onBlur={[Function]}
            type="number"
          />
        </label>
      </li>
    </ul>
    <div
      className="form_btn_block"
    >
      <button
        className="clients_btn save_btn"
        type="submit"
      >
        save
      </button>
      <button
        className="clients_btn cancel_btn"
        onClick={[Function]}
        type="button"
      >
        cancel
      </button>
    </div>
  </form>
</div>
`;
